### ISSUE TITLE
How to refer properly to fallback image?

### ISSUE THREAD

**Issue by anti-destin** [refile/refile#392]
What is the proper way to refer to the fallback image when using `attachment_image_tag`?

I have `avatar_18` in `app/assets/images`. I refer to it as follows:

`<%= attachment_image_tag(current_user.avatars.first, :file, :fill, 18, 18, class: 'img-responsive, img-circle', fallback: asset_path('avatar_18') %>`.

But the image links on Heroku are broken.

The problem seems to be the path.

 - `fallback: image_path('avatar_18')` works in development, but not in production; the source is wrong: `src="/images/avatar_18"`.

 - `fallback: asset_path('avatar_18')` works in development, but not in production; the source is wrong: `src="/avatar_18"`. The path is missing `assets`. Also, the digest is missing.

 - Running `ls public/assets` shows the file `avatar_18-3c7e2d23b8adbdf979eff7152c156b18817f26b4ceae7525dcdded75d523a052.png`.

 - I have also tried checking the asset_path: `irb(main):001:0> puts helper.asset_path('avatar_18.png')
    /assets/avatar_18-3c7e2d23b8adbdf979eff7152c156b18817f26b4ceae7525dcdded75d523a052.png
    => nil`.

Is this a Rails issue, a Heroku issue, or a Refile issue? What else should I check?

**Comment by wkurosawa** [comment]
I used just the name of the file, without `asset_path` helper and it's working.
Can you try? 
`attachment_image_tag(event, :main_image, :fill, 250, 150, fallback: 'logo.png')`

**Comment by anti-destin** [comment]
Sorry, I should have clarified that that is what I had initially, but it did not work, which is why I tried using a helper.

In particular, `<%= attachment_image_tag(current_user.avatars.first, :file, :fill, 18, 18, class: 'img-responsive, img-circle', fallback: 'avatar_18') %>` generates `<img class="attachment file img-responsive, img-circle fallback" src="/images/avatar_18" alt="Avatar 18">` on Heroku.

The first problem is that `images` should be `assets`, and the second problem is that the digest is missing.

Were you working in development or production?

**Comment by wkurosawa** [comment]
I see.. 
That's my production environment, but it's a project using refile `0.5.4`, without `Refile.cdn_host` configuration.

What's your refile version?

**Comment by anti-destin** [comment]
From `bundle show`: refile (0.5.5 0dc2072).

Does the digest show up for you?

**Comment by wkurosawa** [comment]
Yes, 

```ruby
attachment_image_tag(user, :profile_image, :fill, width, height, format: "jpg", fallback: "logo.png")
```

generates

```html
<img class="attachment user profile_image fallback" src="/assets/logo-5e80df06d38d0486934e39a807e9d2001c379387f5de6791354a847e4f5305a0.png" alt="Logo 5e80df06d38d0486934e39a807e9d2001c379387f5de6791354a847e4f5305a0">
```

**Comment by anti-destin** [comment]
Strange. Do you use Heroku or some other provider? Not sure what else to check.

**Comment by wkurosawa** [comment]
Ah, yes, I'm using Heroku and S3 backend. 
Back in 0.5 version S3 was included in refile gem. 
Are you using S3? 
And have you tried with file extension? Guess it doesn't matter though..

**Comment by anti-destin** [comment]
Yes, I use S3 also. But anyway, the fallback image should be served from `/assets/`. And the file extension should not matter.

This is why I was wondering whether there was a helper that I could use to refer to the image explicitly. But none seems to work.

**Comment by wkurosawa** [comment]
Really weird. `attachment_image_tag` uses rails `image_tag` helper to generate fallback's code: https://github.com/refile/refile/blob/v0.5.5/lib/refile/rails/attachment_helper.rb#L47
It should work as any `image_tag` instruction.

**Comment by anti-destin** [comment]
You were right! I had to add the file extension when indicating the fallback image. Not having the file extension is fine in development, but breaks in production. I guess that this is why I missed it. Thanks for your help on this.

**Comment by wkurosawa** [comment]
Hooray! You're welcome! 
Can you close this issue by yourself?
